src/pyskindose/plotting/create_dose_map_plot.py

```diff
@@ -1,21 +1,256 @@
+        settings: PyskindoseSettings,
+        dose_map: np.ndarray) -> None:
+    """Plot a map of the absorbed skindose upon the patient phantom.
+
+    This function creates and plots an offline plotly graph of the
+    skin dose distribution on the phantom. The colorscale is mapped to the
+    absorbed skin dose value. Only available for phantom type: "plane",
+    "cylinder" or "human"
+
+    Parameters
+    ----------
+    patient : Phantom
+        patient phantom
+    settings : PyskindoseSettings
+        Settings class for PySkinDose
+    dose_map : np.ndarray
+        array with dose matrix, where each element is to be mapped to the
+        corresponding skin cell of the patient.
+
+    """
     # Fix error with plotly layout for 2D plane patient.
-    if patient.phantom_model == c.PHANTOM_MODEL_PLANE:
+    if patient.phantom_model == PHANTOM_MODEL_PLANE:
         patient = Phantom(
-            phantom_model=settings.phantom.model, phantom_dim=settings.phantom.dimension
-        )
+            phantom_model=settings.phantom.model,
+            phantom_dim=settings.phantom.dimension
+            )
 
-    # Append and plot dose map
+    # append dosemap to patient
     patient.dose = dose_map
-    patient.plot_dosemap(
-        dark_mode=settings.plot.dark_mode,
+
+    COLOR_CANVAS, COLOR_PLOT_TEXT, COLOR_GRID, COLOR_ZERO_LINE = \
+        fetch_plot_colors(dark_mode=settings.plot.dark_mode)
+
+    PLOT_HEIGHT, PLOT_WIDTH = fetch_plot_size(
         notebook_mode=settings.plot.notebook_mode)
+
+    PLOT_MARGINS = fetch_plot_margin(notebook_mode=settings.plot.notebook_mode)
+
+    lat_text = [f"<b>lat:</b> {np.around(patient.r[ind, 2],2)} cm<br>"
+                for ind in range(len(patient.r))]
+
+    lon_text = [f"<b>lon:</b> {np.around(patient.r[ind, 0],2)} cm<br>"
+                for ind in range(len(patient.r))]
+
+    ver_text = [f"<b>ver:</b> {np.around(patient.r[ind, 1],2)} cm<br>"
+                for ind in range(len(patient.r))]
+
+    dose_text = [f"<b>skin dose:</b> {round(patient.dose[ind],2)} mGy"
+                 for ind in range(len(patient.r))]
+
+    hover_text = [lat_text[cell] + lon_text[cell] + ver_text[cell] +
+                  dose_text[cell] for cell in range(len(patient.r))]
+
+    # create mesh object for the phantom
+    phantom_mesh = [
+        go.Mesh3d(
+            x=patient.r[:, 0], y=patient.r[:, 1], z=patient.r[:, 2],
+            i=patient.ijk[:, 0], j=patient.ijk[:, 1], k=patient.ijk[:, 2],
+            intensity=patient.dose, colorscale=DOSEMAP_COLORSCALE,
+            showscale=True,
+            hoverinfo='text',
+            text=hover_text, name="Human",
+            colorbar=dict(tickfont=dict(color=COLOR_PLOT_TEXT),
+                          title="Skin dose [mGy]",
+                          titlefont=dict(
+                            family=PLOT_FONT_FAMILY,
+                            color=COLOR_PLOT_TEXT)))]
+
+    # Layout settings
+    layout = go.Layout(
```

Comment:
Could we extract the layout creation to a separate function to make this one more lean?

Reply:
yes, resolved in a1dfe919b9f1549a96b9951c1c250b86722f2613